### ISSUE TITLE
get rid of <[Object: null prototype] {}> console noise

### ISSUE THREAD

**Issue by erights** [Agoric/agoric-sdk#2324]
After applying both https://github.com/Agoric/agoric-sdk/pull/2303 and https://github.com/Agoric/SES-shim/pull/569 , a `yarn test` fails in something having to do with xsnap, complaining about an assignment to a `constructor` property failing.

***Low priority*** because https://github.com/Agoric/SES-shim/pull/569 is low priority. But fixing this would cause the Node console to stop adding junk like the following to its display of objects.
```js
[Function (anonymous)] Function <Function <[Object: null prototype] {}>>
```

**Comment by erights** [comment]
When linking `master` of `agoric-sdk` with https://github.com/Agoric/SES-shim/pull/569` i.e., https://github.com/Agoric/SES-shim/tree/markm-stop-enabling-O-p-constructor and then doing a `yarn test` I get lots of error output in which this seems to be the key diagnostic:

```
    message: 'Cannot assign to read only property \'constructor\' of object \'[object Object]\' in /Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/whitelist.js',
```

However, this is unlikely to be a problem in the source code of `whitelist.js` since it doesn't have this problem otherwise. But perhaps?

This diagnostic appears within the following longer sample

```
Replaying SwingSet transcripts
Replaying SwingSet transcripts
  devices › before hook

  Rejected promise returned by test. Reason:

  TypeError {
    code: 'PLUGIN_ERROR',
    hook: 'transform',
    id: '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/whitelist.js',
    plugin: 'commonjs',
    watchFiles: [
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/SwingSet/src/kernel/vatManager/lockdown-subprocess-xsnap.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/xsnap/lib/bootstrap.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/xsnap/lib/console-shim.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/xsnap/lib/text-shim.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/xsnap/lib/lockdown-shim.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/eventual-send/shim.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/lockdown.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/eventual-send/src/index.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/commons.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/tame-function-tostring.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/intrinsics.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/get-anonymous-intrinsics.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/lockdown-shim.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/compartment-shim.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/assert.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/eventual-send/src/track-turns.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/whitelist.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/inert.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/whitelist-intrinsics.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/repair-legacy-accessors.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/tame-function-constructors.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/tame-date-constructor.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/tame-math-object.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/tame-regexp-constructor.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/enable-property-overrides.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/tame-locale-methods.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/global-object.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/evaluate.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/scope-constants.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/transforms.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/tame-console.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/tame-error-constructor.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/stringify-utils.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/types.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/internal-types.js',
    ],
    message: 'Cannot assign to read only property \'constructor\' of object \'[object Object]\' in /Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/whitelist.js',
  }



  demos › run encouragementBot Demo

  Rejected promise returned by test. Reason:

  TypeError {
    code: 'PLUGIN_ERROR',
    hook: 'transform',
    id: '/Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/whitelist.js',
    plugin: 'commonjs',
    watchFiles: [
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/SwingSet/src/kernel/vatManager/lockdown-subprocess-xsnap.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/xsnap/lib/bootstrap.js',
      '/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/xsnap/lib/console-shim.js',
```

**Comment by erights** [comment]
Fixing `Object.prototype.constructor` would also remove two of the eight junk lines in the VSCode debugger's object inspector of all objects. This is a significant noise reduction of the debugging experience.

**Comment by erights** [comment]
Sorry closed by mistake

**Comment by dckc** [comment]
with `stackFiltering` set to `verbose` in `install-ses` I get more detail:

```

@@@@@@2 [TypeError: Cannot assign to read only property 'constructor' of object '[object Object]' in /home/connolly/projects/agoric/SES-shim/packages/ses/src/whitelist.js
  at Object.pp$2.checkLVal (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:1814:33)
  at Object.pp$3.checkParams (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:2717:12)
  at Object.pp$3.parseFunctionBody (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:2685:12)
  at Object.pp$1.parseFunction (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:1283:10)
  at Object.pp$1.parseFunctionStatement (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:983:17)
  at Object.pp$1.parseStatement (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:830:19)
  at Object.parseStatement (/home/connolly/projects/agoric/agoric-sdk/packages/bundle-source/node_modules/rollup/dist/shared/node-entry.js:11825:30)
  at Object.pp$1.parseTopLevel (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:746:23)
  at Object.parse (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:553:17)
  at InertConstructor.parse (/home/connolly/projects/agoric/agoric-sdk/node_modules/acorn/dist/acorn.js:576:37)
  at Graph.contextParse (/home/connolly/projects/agoric/agoric-sdk/packages/bundle-source/node_modules/rollup/dist/shared/node-entry.js:13216:70)
  at tryParse (/home/connolly/projects/agoric/agoric-sdk/packages/bundle-source/node_modules/@rollup/plugin-commonjs/dist/index.js:282:12)
  at checkEsModule (/home/connolly/projects/agoric/agoric-sdk/packages/bundle-source/node_modules/@rollup/plugin-commonjs/dist/index.js:296:15)
  at Object.transformAndCheckExports (/home/connolly/projects/agoric/agoric-sdk/packages/bundle-source/node_modules/@rollup/plugin-commonjs/dist/index.js:801:28)
  at Object.transform (/home/connolly/projects/agoric/agoric-sdk/packages/bundle-source/node_modules/@rollup/plugin-commonjs/dist/index.js:884:48)
  at /home/connolly/projects/agoric/agoric-sdk/packages/bundle-source/node_modules/rollup/dist/shared/node-entry.js:13113:25] {
  id: '/home/connolly/projects/agoric/SES-shim/packages/ses/src/whitelist.js',
  hook: 'transform',
  code: 'PLUGIN_ERROR',
  plugin: 'commonjs'
}
```

**Comment by dckc** [comment]
I'm expanding the scope of this to actually getting rid of the console noise; i.e. doing the SES-shim release @erights asked for and upgrading agoric-sdk to use it.

**Comment by erights** [comment]
> I'm expanding the scope of this to actually getting rid of the console noise; i.e. doing the SES-shim release @erights asked for and upgrading agoric-sdk to use it.

Excellent, thanks!

If we can get https://github.com/Agoric/SES-shim/pull/590 and https://github.com/Agoric/SES-shim/pull/591 in quickly enough to make this release, that would be nice. Otherwise they can wait until the next release.

**Comment by dckc** [comment]
#2552 failed `lint:types` due to outdated `@agoric/make-hardener` types. I tried to make a release with the types fixed (9b16d93fe94b9) but I don't have write access to https://www.npmjs.com/package/@agoric/make-hardener .

I wouldn't mind keeping it that way :)

@warner perhaps you're willing to move the ball ahead the next few yards? If not, nominate somebody? (if it's me, grant me npm write access)
